presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformCorrelatedScalarSubquery.java

```diff
@@ -131,10 +131,12 @@ public Result apply(LateralJoinNode lateralJoinNode, Captures captures, Context
                 lateralJoinNode.getOriginSubquery());
 
         Symbol isDistinct = context.getSymbolAllocator().newSymbol("is_distinct", BooleanType.BOOLEAN);
```

Comment:
Given we need to do this conversion very frequently, add a utility function in symbolAllocator to get VariableReferenceExpression?
```
context.getSymbolAllocator().newSymbol(...);
context.getSymbolAllocator().getVariableReference(symbol);
```

Reply:
It doesn't really matter. However you write it now, eventually these will be changed again. None of these will be here once the refactor is done.